javascript_nodejs/50.contoso-cafe-bot/bot.js

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes, CardFactory, MessageFactory } = require('botbuilder');
+const { DialogTurnStatus, DialogSet } = require('botbuilder-dialogs');
+const { LuisRecognizer } = require('botbuilder-ai');
+const { OnTurnProperty } = require('./dialogs/shared/stateProperties');
+const { MainDialog } = require('./dialogs/main');
+const { WelcomeCard } = require('./dialogs/welcome');
+
+// LUIS service type entry in the .bot file for dispatch.
+const LUIS_CONFIGURATION = 'cafeDispatchModel';
+
+// State properties
+const ON_TURN_PROPERTY = 'onTurnStateProperty';
+const DIALOG_STATE_PROPERTY = 'dialogStatePropety';
+
+module.exports = {
+    Bot: class {
+        /**
+         * Bot constructor.
+         * 
+         * @param {Object} conversationState conversation state object
+         * @param {Object} userState user state object
+         * @param {Object} botConfig bot configuration
+         * 
+         */
+        constructor (conversationState, userState, botConfig) {
+            if (!conversationState) throw ('Missing parameter. Conversation state is required.');
+            if (!userState) throw ('Missing parameter. User state is required.');
+            if (!botConfig) throw ('Missing parameter. Bot configuration is required.');
+
+            // Create state property accessors.
+            this.onTurnPropertyAccessor = conversationState.createProperty(ON_TURN_PROPERTY);
+            this.dialogPropertyAccessor = conversationState.createProperty(DIALOG_STATE_PROPERTY);
+            
+            // add recogizers
+            const luisConfig = botConfig.findServiceByNameOrId(LUIS_CONFIGURATION);
+            if (!luisConfig || !luisConfig.appId) throw (`Cafe Dispatch LUIS model not found in .bot file. Please ensure you have all required LUIS models created and available in the .bot file. See readme.md for additional information.\n`);
+            this.luisRecognizer = new LuisRecognizer({
+                applicationId: luisConfig.appId,
+                azureRegion: luisConfig.region,
```

Comment:
> azureRegion: luisConfig.region, [](start = 15, length = 32)

I don't think this is the signature anymore.  Isn't this endpoint?  I think you want:
endpoint: luisConfig.getEndpoint()

Reply:
Will fix in all up samples clean up next week.